install/management/appliance.go

```diff
@@ -348,12 +330,19 @@ func (d *Dispatcher) setMacToGuestInfo() error {
 			err = errors.Errorf("Timeout to get VM MAC address")
 			return err
 		}
-		netName := d.nics[key]
-		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/portgroup", netName), Value: key})
-		spec.ExtraConfig = append(spec.ExtraConfig, &types.OptionValue{Key: fmt.Sprintf("guestinfo.vch/networks/%s/mac", netName), Value: value})
-		keys = append(keys, netName)
+		for nicName, netInfo := range conf.Networks {
```

Comment:
@hickeng the map conf.Networks is one mapping between nic name and networkinfo, but mac address mapping is between network name and mac address. 
The key is different, that's why I loop the map and compare there key with following sentence:
		for nicName, netInfo := range conf.Networks {
			if netInfo.PortGroupName == netName {
You can see that's not the key comparison.

But you do find an issue for the mac address waiting. I assume we should wait until mac address of all nics are ready, instead of any of them is there. So I changed the WaitForMAC method.